kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -70,74 +72,77 @@ function _collectionState(data) {
-        }
-        resolve(returnChannelList);
-      });
-  });
+function _channelListState(data) {
+  return data.map(channel => ({
+    id: channel.id,
+    title: channel.name,
+    description: channel.description,
+    root_id: channel.root_pk,
+  }));
 }
 
 
-/*
- * Returns a promise that fetches the current channel id.
+/**
+ * Title Helpers
  */
-function _getCurrentChannelId() {
-  let currentChannelId = null;
-  return new Promise((resolve, reject) => {
-    _getChannelList()
-      .then((channelList) => {
-        if (channelList && channelList.length) {
-          const cookieCurrentChannelId = cookiejs.get('currentChannel');
-          if (channelList.some((channel) => channel.id === cookieCurrentChannelId)) {
-            currentChannelId = cookieCurrentChannelId;
-          } else {
-            currentChannelId = channelList[0].id;
-          }
-        }
-        resolve(currentChannelId);
-      });
-  });
+
+function _explorePageTitle(title) {
+  if (title) {
+    return `Explore - ${title}`;
+  }
+  return 'Explore';
+}
+
+function _learnPageTitle(title) {
+  if (title) {
+    return `Learn - ${title}`;
+  }
+  return 'Learn';
+}
+
+function _errorTitle(title) {
+  if (title) {
+    return `Error - ${title}`;
+  }
+  return 'Error';
 }
 
 
 /*
- * Returns a promise that fetches the the root topic id of the current channel.
+ * Returns the 'default' channel ID:
+ * - if there are channels and they match the cookie, return that
+ * - else if there are channels, return the first one
+ * - else return null
  */
-function _getCurrentChannelRootTopicId() {
-  let currentChannelRootTopicId = null;
-  return new ConditionalPromise((resolve, reject) => {
-    const currentChannelIdPromise = _getCurrentChannelId();
-    const channelListPromise = _getChannelList();
-    Promise.all([currentChannelIdPromise, channelListPromise])
-    .then(([currentChannelId, channelList]) => {
-      if (currentChannelId && channelList) {
-        for (const channel in channelList) {
-          if (channelList[channel].id === currentChannelId) {
-            currentChannelRootTopicId = channelList[channel].root_pk;
-            break;
-          }
-        }
-      }
-      resolve(currentChannelRootTopicId);
-    });
-  });
+function _getDefaultChannelId(store, channelList) {
+  if (channelList && channelList.length) {
+    const cookieVal = cookiejs.get('currentChannelId');
+    if (channelList.some((channel) => channel.id === cookieVal)) {
+      return cookieVal;
+    }
+    return channelList[0].id;
+  }
+  return null;
 }
 
-function _updateChannelList(store) {
-  const channelPromise = _getChannelList();
-  channelPromise.then((channelList) => {
-    store.dispatch('SET_CHANNEL_LIST', channelList);
-  });
-  return channelPromise;
+
+/*
+ * Set channel state info.
+ * Returns `true` if the requested channel ID is available in the list.
+ */
+function _handleChannels(store, currentChannelId, channelList) {
```

Comment:
Not entirely comfortable with its combined use as a check, given that it has side effects. Could we have one function that just does the check, and another that is an action that does the mutation dispatch?